src/storage/btree.c

```diff
@@ -16683,12 +16955,20 @@ btree_get_next_key_info (THREAD_ENTRY * thread_p, BTID * btid, BTREE_SCAN * bts,
   if (bts->C_page != NULL)
     {
       LSA_COPY (&bts->cur_leaf_lsa, pgbuf_get_lsa (bts->C_page));
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+      btree_check_decompress_key (bts);
+#endif
       pgbuf_unfix_and_init (thread_p, bts->C_page);
     }
+  else
+    {
+      /* No current page fixed. */
+      LSA_SET_NULL (&bts->cur_leaf_lsa);
+    }
 
   if (bts->O_page != NULL)
     {
-      pgbuf_unfix_and_init (thread_p, bts->C_page);
+      pgbuf_unfix_and_init (thread_p, bts->O_page);
```

Comment:
😱

Reply:
This is not what we want to cover in this issue, but since it was such an obvious error, we fixed it together.

There may have been a memory leak in the past, but since it is only called when using the **SELECT_KEY_INFO** hint, it is unlikely to occur often on actual sites.

Reply:
I think it would be good to hear your opinion.

Reply:
I will not cover this matter here, but will create a separate issue to handle it.

Reply:
For backport purposes, it would be better to resolve this issue in another issue.
I am shocked that there was a typo like this until now.